osquery/bro/BrokerManager.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <broker/broker.hh>
+#include <broker/endpoint.hh>
+#include <broker/message_queue.hh>
+
+#include "osquery/bro/QueryManager.h"
+
+#include <osquery/database.h>
+#include <osquery/status.h>
+#include <osquery/system.h>
+
+#include <algorithm>
+#include <iostream>
+#include <list>
+
+namespace osquery {
+
+class BrokerManager {
+ private:
+  BrokerManager();
+
+ public:
+  // Get a singleton instance
+  static BrokerManager* getInstance() {
```

Comment:
We have a few examples of singleton use, check out the `Config` or `RegistryFactory` classes for best-practice.